src/ui/canvas/gdi/Bitmap.cpp

```
@@ -22,36 +22,64 @@ Copyright_License {
 */
 
 #include "ui/canvas/Bitmap.hpp"
+#include "ui/canvas/gdi/GdiPlusBitmap.hpp"
+
 #include "Screen/Debug.hpp"
 #include "system/Path.hpp"
 
 #include <cassert>
 
+#if !defined(USE_OPENGL) && !defined(USE_MEMORY_CANVAS)
+std::unordered_map<HBITMAP, size_t> Bitmap::bitmap_map;
+#endif
 Bitmap::Bitmap(Bitmap &&src)
-  :bitmap(src.bitmap)
+  :hbitmap(src.hbitmap)
 {
-  src.bitmap = nullptr;
+
+  if (src.IsDefined()) {
+  auto pair = bitmap_map.find(src.hbitmap);
+  if (pair != bitmap_map.end()) {
+    pair->second++;
+  }
+  }
+
 }
 
 bool
 Bitmap::LoadFile(Path path)
 {
-  return false;
+  hbitmap = GdiLoadImage(path.c_str());
+  if (!IsDefined()) {
+    LogFormat(_T("Bitmap couldn't load:  %d!"), GetLastError());
```

Throw exception instead of logging